hugegraph-server/hugegraph-core/pom.xml

```
@@ -198,7 +198,7 @@
             <version>${commons-compress.version}</version>
         </dependency>
         <dependency>
-            <groupId>org.lz4</groupId>
+            <groupId>at.yawk.lz4</groupId>
             <artifactId>lz4-java</artifactId>
```

‼️ **Critical: Incorrect GroupId for lz4-java**

The dependency change switches from `org.lz4:lz4-java` to `at.yawk.lz4:lz4-java`, but this appears to be incorrect. The official lz4-java artifact is maintained under the `org.lz4` groupId.

The `at.yawk.lz4` groupId points to an unofficial fork that may not be maintained and could introduce compatibility or security issues.

**Recommendation:** Keep the original `org.lz4` groupId and only update the version to 1.9.0:

```suggestion
        <dependency>
            <groupId>org.lz4</groupId>
            <artifactId>lz4-java</artifactId>
            <version>${lz4.version}</version>
        </dependency>
```

---

You are correct, the official latest version, 1.8.1, also includes the security implementation fix. I will switch to using version 1.8.1 from the official Group ID org.lz4.